rflx/parser/cache.py

```diff
@@ -0,0 +1,63 @@
+import hashlib
+import json
+import pathlib
+from typing import Dict
+
+from rflx import __version__
+from rflx.model.message import AbstractMessage
+
+CACHE_DIR = pathlib.Path.home() / ".cache" / "RecordFlux"
+VERIFICATION_FILE = "verification.json"
+
+
+class Cache:
+    def __init__(self, enabled: bool = True) -> None:
+        self._enabled = enabled
+
+        if not enabled:
+            return
+
+        self._verification: Dict[str, str] = {}
+
+        self._initialize_cache()
+        self._load_cache()
+
+    def is_verified(self, message: AbstractMessage) -> bool:
+        if not self._enabled:
+            return False
+
+        return message.full_name in self._verification and self._verification[
+            message.full_name
+        ] == self._message_hash(message)
+
+    def add_verified(self, message: AbstractMessage) -> None:
+        if not self._enabled:
+            return
+
+        message_hash = self._message_hash(message)
+        if (
+            message.full_name not in self._verification
+            or message_hash != self._verification[message.full_name]
+        ):
+            self._verification[message.full_name] = message_hash
+            self._write_cache()
+
+    @staticmethod
+    def _initialize_cache() -> None:
+        if not CACHE_DIR.exists():
+            CACHE_DIR.mkdir()
```

Comment:
This will fail if `$HOME/.cache` does not exist. You could use `CACHE_DIR.mkdir(parents=True)`.

Reply:
Fixed by 6c59a81.